jax_unirep/evotuning.py

```diff
@@ -27,6 +26,53 @@
     validate_mLSTM1900_params,
 )
 
+# HERE LIES THE DRAG.. MODEL!
+"""
+init_fun, predict = serial(
+    mLSTM1900(), mLSTM1900_HiddenStates(), Dense(25), Softmax
+)
+"""
+# I hypothesized the Softmax layer wasn't working
+# --> Commented out and showed it made no difference!
+# Now I'll just implement softmax manually.
+init_fun, predict = serial(mLSTM1900(), mLSTM1900_HiddenStates(), Dense(25))
+
+
+def softmax(y_hat):
+    """
+    Returns a 3D array of the same dimensions but with softmax applied to the probabilities,
+    to make them probabilities!
+    :param y_hat: A 3D array of dimensions (# sequences, # AA's, 25),
+        input sequences will be batched to have consistent # AA's,
+        & 25 represents a probability of each possible AA.
+
+    """
+    return np.exp(y_hat) / np.sum(np.exp(y_hat), axis=2, keepdims=True)
```

Comment:
If this implementation proves to be working, it should be moved to `activations.py`.